frontend/src/metabase/visualizations/lib/settings/column.unit.spec.js

```diff
@@ -95,4 +104,98 @@ describe("column settings", () => {
     expect(time_style).toEqual("h:mm A");
     expect(date_style).toEqual("");
   });
+
+  it("should show new columns and show them as disabled (metabase#25592)", () => {
+    const storedSettings = {
+      "table.columns": [
+        createMockTableColumnOrderSetting({
+          name: "ID",
+          fieldRef: ["field", ORDERS.ID, null],
+          enabled: true,
+        }),
+      ],
+    };
+
+    const series = [
+      createMockSingleSeries(
+        {},
+        {
+          data: {
+            cols: [
+              createMockColumn({
+                id: ORDERS.ID,
+                name: "ID",
+                display_name: "Id",
+                field_ref: ["field", ORDERS.ID, null],
+              }),
+              createMockColumn({
+                id: ORDERS.SUBTOTAL,
+                name: "SUBTOTAL",
+                display_name: "Total",
+                field_ref: ["field", ORDERS.SUBTOTAL, null],
+              }),
+            ],
+          },
+        },
+      ),
+    ];
+
+    const computedValue = buildTableColumnSettings()["table.columns"].getValue(
+      series,
+      storedSettings,
+    );
+
+    const computedSubtotal = computedValue.find(
+      ({ name }) => name === "SUBTOTAL",
+    );
+
+    expect(computedSubtotal).not.toBeUndefined();
+    expect(computedSubtotal.enabled).toBe(false);
+  });
+
+  it("table should should generate default columns when table.columns entries do not match data.cols (metabase#28304)", () => {
```

Comment:
I trasformed `e2e/test/scenarios/visualizations/reproductions/28304-table-columns-unknown.cy.spec.js` into a unit test to make it run faster